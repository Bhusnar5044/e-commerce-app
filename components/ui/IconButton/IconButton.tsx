import { memo } from 'react';

import { cn } from '@utils';
import { IconButtonProps } from './types';

export const IconButton: React.FC<IconButtonProps> = memo(({ onClick, icon, className }) => {
    return (
        <button
            title="iconButton"
            onClick={onClick}
            className={cn(
                'rounded-full flex items-center justify-center bg-white border shadow-md p-2 hover:scale-110 transition',
                className,
            )}
        >
            {icon}
        </button>
    );
});

IconButton.displayName = 'IconButton';
