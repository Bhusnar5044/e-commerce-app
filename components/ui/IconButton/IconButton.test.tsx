import { render } from '@test-utils';
import { IconButton } from './IconButton';

describe('IconButton', () => {
    it('should render properly', () => {
        const { container } = render(<IconButton icon={<svg />} />);
        expect(container).toMatchSnapshot();
    });
});
