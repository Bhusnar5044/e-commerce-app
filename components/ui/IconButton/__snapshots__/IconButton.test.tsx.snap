// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should render properly 1`] = `
<div>
  <button
    class="rounded-full flex items-center justify-center bg-white border shadow-md p-2 hover:scale-110 transition"
    title="iconButton"
  >
    <svg />
  </button>
</div>
`;
